src/neoxp/Node/ExpressOracle.cs

```diff
@@ -138,5 +139,61 @@ public static void SignOracleResponseTransaction(ProtocolSettings settings, Expr
 
             return tx;
         }
+
+        public const byte Prefix_Contract = 8;
+        private const byte Prefix_NextAvailableId = 15;
+
+        private static int LastUsedContractId(DataCache snapshot)
+        {
+            StorageKey key = new KeyBuilder(NativeContract.ContractManagement.Id, Prefix_NextAvailableId);
+            StorageItem item = snapshot.TryGet(key);
+            return (int)(BigInteger)item;
+        }
+
+        private static void SetLastUsedContractId(DataCache snapshot, int newId)
+        {
+            StorageKey key = new KeyBuilder(NativeContract.ContractManagement.Id, Prefix_NextAvailableId);
+            StorageItem item = snapshot.GetAndChange(key);
+            item.Set(newId);
+        }
+        
+        private static int GetNextAvailableId(DataCache snapshot)
+        {
+            StorageKey key = new KeyBuilder(NativeContract.ContractManagement.Id, Prefix_NextAvailableId);
+            StorageItem item = snapshot.GetAndChange(key);
+            int value = (int)(BigInteger)item;
+            item.Add(1);
+            return value;
+        }
+        
+        public static bool PersistContract(SnapshotCache snapshot, ContractState state, (byte[] key, byte[] value)[] storagePairs)
+        {
+            // Our local chain might already be using the contract id of the pulled contract, we need to check for this
+            // to avoid having contracts with duplicate id's. This is important because the contract id is part of the
+            // StorageContext used with Storage syscalls and else we'll potentially override storage keys or iterate
+            // over keys that shouldn't exist for one of the contracts.
+            if (state.Id <= LastUsedContractId(snapshot))
+            {
+                state.Id = GetNextAvailableId(snapshot);
+            }
+            else
+            {
+                // Update available id such that a regular contract deploy will use the right next id;
+                SetLastUsedContractId(snapshot, state.Id);
+            }
+            
+            StorageKey key = new KeyBuilder(NativeContract.ContractManagement.Id, Prefix_Contract).Add(state.Hash);
```

Comment:
What happens if the contract already exists? We need some type of `force` argument so the developer doesn't overwrite important data by accident

Reply:
I think that's part of the discussion here https://github.com/neo-project/neo-express/pull/209#discussion_r826882935
As said on Discord I want to get the base in first and after that deal with the optional download parameters.

Reply:
I'm fine getting the base line code working and approved and then dealing with handling overwriting. Neo platform is planning to ship [v3.2 release](https://github.com/neo-project/neo/issues/2676) in a couple weeks. As long as we get `contract download` in before that, I don't think we're in a rush otherwise

Reply:
We can still do it in this PR. Just doing it last of all